pkg/volume/flexvolume/probe.go

```diff
@@ -25,31 +25,26 @@ import (
 	"fmt"
 	"path/filepath"
 	"sync"
-	"time"
+
+	"strings"
 
 	"github.com/fsnotify/fsnotify"
 	"k8s.io/apimachinery/pkg/util/errors"
 	utilfs "k8s.io/kubernetes/pkg/util/filesystem"
+	utilstrings "k8s.io/kubernetes/pkg/util/strings"
 )
 
 type flexVolumeProber struct {
 	mutex           sync.Mutex
 	pluginDir       string // Flexvolume driver directory
 	watcher         utilfs.FSWatcher
-	probeNeeded     bool      // Must only read and write this through testAndSetProbeNeeded.
-	lastUpdated     time.Time // Last time probeNeeded was updated.
 	watchEventCount int
 	factory         PluginFactory
 	fs              utilfs.Filesystem
+	probeAllNeeded  bool
+	eventsMap       map[string]volume.ProbeEvent
 }
 
-const (
-	// TODO (cxing) Tune these params based on test results.
-	// watchEventLimit is the max allowable number of processed watches within watchEventInterval.
-	watchEventInterval = 5 * time.Second
```

Comment:
for two reasons:
1. `watchEventInterval ` is used in func `updateProbeNeeded()`, but `probeNeeded` is removed
2.  In my approach, we can't miss any event, because `probe()` will not scan the whole pluginDir except the first time it is invoked

Reply:
I thought for a long time, and can't find any remedy after missing some event. I think we only need to consider the normal situation, malicious attacks as a security issue should be resolved from a higher level.